mmr-lib/Models/Settings/Settings.cs

```diff
@@ -0,0 +1,323 @@
+        /// </summary>
+        public bool OutputN64ROM
+        {
+            get => outputN64ROM;
+            set => SetField(ref outputN64ROM, value);
+        }
+        private bool outputN64ROM = true;
+
+
+        /// <summary>
+        ///  Outputs virtual channel if true
+        /// </summary>
+        public bool OutputVC
+        {
+            get => outputVC;
+            set => SetField(ref outputVC, value);
+        }
+        private bool outputVC;
+
+        /// <summary>
+        /// Filepath to the input ROM
+        /// </summary>
+        public string InputROMFilename
+        {
+            get => inputROMFilename;
+            set => SetField(ref inputROMFilename, value);
+        }
+        private string inputROMFilename;
+
+        /// <summary>
+        /// Filepath to the input patch file
+        /// </summary>
+        public string InputPatchFilename
+        {
+            get => inputPatchFilename;
+            set => SetField(ref inputPatchFilename, value);
+        }
+        private string inputPatchFilename;
+
+        /// <summary>
+        /// Filepath to the input logic file
+        /// </summary>
+        public string UserLogicFileName
+        {
+            get => userLogicFileName;
+            set => SetField(ref userLogicFileName, value);
+        }
+        private string userLogicFileName;
+
+        /// <summary>
+        /// Default Filename for the output ROM
+        /// </summary>
+        public string DefaultOutputROMFilename
+        {
+            get
+            {
+                return $"MMR_{Seed}.z64";
+            }
+        }
+
+        /// <summary>
+        /// Filepath to the output ROM
+        /// </summary>
+        public string OutputROMFilename
+        {
+            get => outputROMFilename;
+            set => SetField(ref outputROMFilename, value);
+        }
+        private string outputROMFilename;
+
+        /// <summary>
+        /// Generate spoiler log on randomizing
+        /// </summary>
+        public bool OutputTextSpoiler
+        {
+            get => outputTextSpoiler;
+            set
+            {
+                SetField(ref outputTextSpoiler, value);
+            }
+        }
+        private bool outputTextSpoiler;
+
+        /// <summary>
+        /// Generate HTML spoiler log on randomizing
+        /// </summary>
+        public bool OutputHTMLSpoiler
+        {
+            get => outputHTMLSpoiler;
+            set => SetField(ref outputHTMLSpoiler, value);
+        }
+        private bool outputHTMLSpoiler = true;
+
+        /// <summary>
+        /// Use Custom Item list for the logic.
+        /// </summary>
+        public bool UseCustomItemList
+        {
+            get => useCustomItemList;
+            set
+            {
+                SetField(ref useCustomItemList, value);
+
+                //Todo: enable/disable ui elements instead, then have a "validate" stage for generating settings
```

Comment:
Creating a "validate" stage right now is not super important because we only have the one gui for now, but technically with the core logic being in it's own library now, it should be more responsible in managing it's own state within the mmr lib, and not the gui.